src/coreComponents/constitutive/fluid/ReactiveBrineFluid.hpp

```diff
@@ -0,0 +1,358 @@
+
+  // 3. Compute phase densities and phase viscosities
+  m_phase.density.compute( pressure,
+                            temperatureInCelsius,
+                            phaseCompFraction.value[0].toSliceConst(), phaseCompFraction.derivs[0].toSliceConst(),
+                            phaseDensity.value[0], phaseDensity.derivs[0],
+                            m_useMass );
+
+  m_phase.viscosity.compute( pressure,
+                             temperatureInCelsius,
+                             phaseCompFraction.value[0].toSliceConst(), phaseCompFraction.derivs[0].toSliceConst(),
+                             phaseViscosity.value[0], phaseViscosity.derivs[0],
+                             m_useMass );
+
+
+  // for now, we have to compute the phase mass density here
+  m_phase.density.compute( pressure,
+                           temperatureInCelsius,
+                           phaseCompFraction.value[0].toSliceConst(), phaseCompFraction.derivs[0].toSliceConst(),
+                           phaseMassDensity.value[0], phaseMassDensity.derivs[0],
+                           true );
+
+  // 4. Compute enthalpy and internal energy
+  if( m_isThermal )
+  {
+    m_phase.enthalpy.compute( pressure,
+                               temperatureInCelsius,
+                               phaseCompFraction.value[0].toSliceConst(), phaseCompFraction.derivs[0].toSliceConst(),
+                               phaseEnthalpy.value[0], phaseEnthalpy.derivs[0],
+                               m_useMass );
+
+    computeInternalEnergy( pressure,
+                           phaseFraction,
+                           phaseMassDensity,
+                           phaseEnthalpy,
+                           phaseInternalEnergy );
+  }
+
+  // 6. Compute total fluid mass/molar density and derivatives
+
+  computeTotalDensity( phaseFraction,
+                       phaseDensity,
+                       totalDensity );
+}
+
+template< typename PHASE >
+GEOSX_HOST_DEVICE inline void
+ReactiveBrineFluid< PHASE >::KernelWrapper::
+  update( localIndex const k,
+          localIndex const q,
+          real64 const pressure,
+          real64 const temperature,
+          arraySlice1d< geosx::real64 const, compflow::USD_COMP - 1 > const & composition ) const
+{
+  
+  computeChemistry( pressure,
+                    temperature,
+                    composition,
+                    m_primarySpeciesConcentration[k],
+                    m_secondarySpeciesConcentration[k],
+                    m_primarySpeciesTotalConcentration[k],
+                    m_kineticReactionRates[k] );
+
+  compute( pressure,
+           temperature,
+           composition,
+           m_phaseFraction( k, q ),
+           m_phaseDensity( k, q ),
+           m_phaseMassDensity( k, q ),
+           m_phaseViscosity( k, q ),
+           m_phaseEnthalpy( k, q ),
+           m_phaseInternalEnergy( k, q ),
+           m_phaseCompFraction( k, q ),
+           m_totalDensity( k, q ) );
```

Comment:
After the PR is merged, we can probably find ways to reduce the code duplication with `CO2BrineFluid` because it has a lot in common with `ReactiveMultiFluid` (the `compute` functions, etc). In the current state of the PR, the `ReactiveMultiFluid` could almost inherit from `CO2BrineFluid`. 

At the same time, if we want a generic reactive capability that would work with true compositional EOS, maybe what I am suggesting above is not the right path (unless `CO2BrineFluid` is refactored to become more general).